### ISSUE TITLE
@include directive

### ISSUE THREAD

**Issue by jccolson** [bats-core/bats-core#308]
Need to include some files to add more contents in a bats file.
If the files aren't sourced, it's possible to declare more tests.
Samples :
`@test "addition using bc" {`
`  result="$(echo 2+2 | bc)"`
`  [ "$result" -eq 4 ]`
`}`
`@include "Tests/*.bats"`

**Comment by jccolson** [comment]
Hello
This is the diff to implement the @include directive.
[Diff.txt](https://github.com/bats-core/bats-core/files/4588604/Diff.txt)

**Comment by Qu4tro** [comment]
Hey :wave: . To create a Pull Request on Github you need to fork the project.

- Click `fork` button
- Add new remote. `git remote add gh https://github.com/jccolson/bats-core` (the link to your forked repo).
- git push (with the options you've used)
- On GH, go to this repo or your repo and you should have a small yellow notice that is suggesting to create a new PR with your newly pushed branch. Otherwise, there's also a button `New Pull Request` or something.

**Comment by martin-schulze-vireso** [comment]
What does this `@include` mechanism offer in comparison to `source`ing another file?

**Comment by jccolson** [comment]
> What does this `@include` mechanism offer in comparison to `source`ing another file?

If you `source`ing another file, bash don't understand @tests directives and bats can't do the corresponding script.
I want to use bats files and not only bash files.

To complete my example, the directory Tests can have the following file :
Tests/Tests.bats :
#!/usr/bin/env bats

@test "addition using dc" {
  result="$(echo 2 2+p | dc)"
  [ "$result" -eq 4 ]
}

The result :
 ✓ addition using bc 
 ✓ addition using dc 

2 tests, 0 failures

**Comment by martin-schulze-vireso** [comment]
I see, but wouldn't `bats Tests/*.bats` achieve the same effect without any changes to the test files at all?

**Comment by sublimino** [comment]
> I see, but wouldn't `bats Tests/*.bats` achieve the same effect without any changes to the test files at all?

This is the approach used internally by Bats it its own test suite FWIW

**Comment by jccolson** [comment]
With the same Tests/Tests.bats and this Tests.bats :
#!/usr/bin/env bats

@test "addition using bc" {
  result="$(echo 2+2 | bc)"
  [ "$result" -eq 4 ]
}

bats Tests/*.bats

[xxx@chappie ~]$ ./Tests.bats
 ✓  
 ✓  
 ✓ addition using bc 
   bats warning: Executed 3 instead of expected 1 tests

1 test, 0 failures

[xxx@chappie ~]$
I can't use bats or source to include tests.

Yes, bats -r Tests.bats Tests/ achieve the same effect but I have more directories and I don't want to remember this or create scripts to launch bats with the right parameters.
And this ?
if [ ! true ]; then
    @include "Tests/*.bats"
fi

**Comment by martin-schulze-vireso** [comment]
I'm having trouble parsing your examples but if I got you correctly, you tried to run `bats Test/*.bats` from inside a .bats file's global scope, which I did not mean to suggest.

> Yes, bats -r Tests.bats Tests/ achieve the same effect but I have more directories and I don't want to remember this or create scripts to launch bats with the right parameters.

So you got for example
```
Tests/
  a.bats
  b.bats
  subfolder/
    c.bats
    d.bats
```

and only want to run `a.bats` and `b.bats` but not `c.bats` and `d.bats`?
This would be achievable with running from the command line (not from within a bats file!) `bats Test/*.bats`.

Additionally, you want to flip running the other tests from within the "entry point" file. What is your rationale for doing that? You want to skip parts of the tests because it takes too long?

Maybe we should start out with an explanation of the fundamental problems you want to solve to be able to vet your suggested implementation of a fix.
